App/Services/PhotoUtils.js

```diff
@@ -1,94 +1,72 @@
 import RNFS from 'react-native-fs'
 import ImageResizer from 'react-native-image-resizer'
 import IPFS from '../../TextileIPFSNativeModule'
 
-export async function queryPhotos () {
-  const latestTimestampString = await AsyncStorage.getItem('latestPhotoTimestamp')
-  const seed = await getPhoto() // Get the newest photo
-  let photos = []
-  if (!latestTimestampString) {
-    // If we've never queried a photo before, just record the newest photo timestamp
-    // or the current time and we'll start from there next time
-    let timestamp = seed ? new Date(seed.photo.timestamp * 1000) : new Date()
-    console.log('NO PREVIOUS PHOTOS QUERIED, SAVING TIMESTAMP:', timestamp.toISOString())
-    await AsyncStorage.setItem('latestPhotoTimestamp', timestamp.toISOString())
-  } else {
-    if (!seed) {
-      return photos
-    }
-    const latestTimestamp = Date.parse(latestTimestampString)
-    let currentPhoto = seed
-    let currentTimestamp = new Date(currentPhoto.photo.timestamp * 1000)
-    while (currentTimestamp > latestTimestamp) {
-      photos.push(currentPhoto.photo)
-      if (!currentPhoto.pageInfo.has_next_page) {
-        break
-      }
-      currentPhoto = await getPhoto(currentPhoto.pageInfo.end_cursor)
-      currentTimestamp = new Date(currentPhoto.photo.timestamp * 1000)
-    }
-    if (photos.length > 0) {
-      const fullDir = RNFS.DocumentDirectoryPath + '/images/full/'
-      const fullExists = await RNFS.exists(fullDir)
-      if (!fullExists) {
-        await RNFS.mkdir(fullDir)
-      }
-      const thumbRelativeDir = 'images/thumb/'
-      const thumbDir = RNFS.DocumentDirectoryPath + '/' + thumbRelativeDir
-      const thumbExists = await RNFS.exists(thumbDir)
-      if (!thumbExists) {
-        await RNFS.mkdir(thumbDir)
-      }
-      for (const photo of photos) {
-        // iOS method
-        if (photo.uri.includes('assets-library://')) {
-          var regex = /[?&]([^=#]+)=([^&#]*)/g, params = {}, match
-          while (match = regex.exec(photo.uri)) {
-            params[match[1]] = match[2]
-          }
-          const path = fullDir + params.id + '.' + 'JPG'
-          await RNFS.copyAssetsFileIOS(photo.uri, path, 0, 0)
-          photo['path'] = path
-          const thumbPath = await resizeImage(photo.path, thumbRelativeDir)
-          photo['thumbPath'] = thumbPath
-        }
-        // Android Method
-        else if (photo.uri.includes('content://media')) {
-          photo['path'] = await IPFS.getFilePath(photo.uri)
-          photo['thumbPath'] = await resizeImage(photo.path, thumbDir)
-        }
+export async function getPage (pageSize, cursor) {
+  let queryParams = { first: pageSize }
+  if (cursor !== null) {
+    queryParams['after'] = cursor
+  }
+  return CameraRoll.getPhotos(queryParams)
+}
+
+export async function getAllPhotos () {
+  let data = []
+  let hasNextPage = true
+  let cursor = null
+  let totalPages = 10 // I wanted to limit the total size of a return array
```

Comment:
Cool, any particular reason? Total edge case, but if you're on a river trip with no cell service and take more than 250 photos, we won't add all your new photos when you come back online.

Reply:
yeah, was thinking through that. there is a performance hit as your existing images grow. to every foreground have to compare 1000s X 1000s arrays. so 250 felt like a comfortable number even for a rafting trip

Reply:
I don't think it's so bad since you changed the code to just have to scan each array one time (one time to map the processed images to an object, and one time to check if each queried photo exists on that processed images object). That's linear performance O(n) and that's as good as it can get.

Reply:
They way it was before was exponential performance overhead, which would have been a problem for sure.

Reply:
upped it